lib/segment/src/vector_storage/chunked_mmap_vectors.rs

```diff
@@ -88,27 +98,18 @@ impl<T: Sized + Copy + 'static> ChunkedMmapVectors<T> {
                 mlock,
                 populate,
             };
-            let mut file = OpenOptions::new()
-                .write(true)
-                .create(true)
-                .truncate(true)
-                .open(&config_file)?;
-            serde_json::to_writer(&mut file, &config)?;
-            file.flush()?;
+            atomic_save_json(&config_file, &config)?;
             Ok(config)
-        } else {
-            let file = std::fs::File::open(&config_file)?;
-            let config: ChunkedMmapConfig = serde_json::from_reader(file)?;
-
-            if config.dim != dim {
-                return Err(OperationError::service_error(format!(
-                    "Wrong configuration in {}: expected {}, found {dim}",
-                    config_file.display(),
-                    config.dim,
-                )));
-            }
+        }
+    }
 
-            Ok(config)
+    fn load_config(config_file: &Path) -> Option<ChunkedMmapConfig> {
+        if config_file.exists() {
+            let file = std::fs::File::open(config_file).ok()?;
+            let config: ChunkedMmapConfig = serde_json::from_reader(file).ok()?;
```

Comment:
I would prefer to keep & log error in case of failure

Reply:
WHYM about keeping? As I remember, the decision was to create a new file if prev one is corrupted

Reply:
Fixed